### PR TITLE
ping: Do not request or save configuration if running a replay

### LINKED ISSUE #343: Application tries to configure file link as a normal sensor

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #343

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
We do not have write steps, also, doing this, this will not make all configuration timers to stop, resulting in a series of warning messages.

**Comment by jaxxzer** [comment]
setLastPingConfiguration is the write step. If the link is not writable, it should be handled inside of that function. You may also handle !iswritable() elsewhere, but it should be at least handled in setLastPingConfiguration.

**Comment by patrickelectric** [comment]
@jaxxzer 	did you check the last commit ?

**Comment by jaxxzer** [comment]
Not before I replied. Will check

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -487,6 +472,29 @@ void Ping::request(int id)
 
 void Ping::setLastPingConfiguration()
 {
+    if(!link()->isWritable()) {
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]:
```suggestion
    static bool once = false;
    if (once) {
        return; // or *disconnect*
    }
    
    if(!link()->isWritable()) {
```

Comment (jaxxzer) [reply src/sensor/ping.cpp]: This will be called on every single message rxed from the device.

Comment (jaxxzer) [reply src/sensor/ping.cpp]: your 'once' did not encapsulate the entire code that needed to be run 'once'. We were trying to set device values from saved settings every time a message was rxed.

**Comment by patrickelectric** [comment]
Fixed.

File: src/sensor/ping.cpp [hunk 2]
```diff
@@ -504,6 +489,35 @@ void Ping::request(int id)
...
+    lastSrcId = _srcId;
+
+    if(!link()->isWritable()) {
+        qCDebug(PING_PROTOCOL_PING) << "It's not possible to set last configuration when link is writable.";
+        return;
+    }
+
+    // Load last configuration
+    static bool once = false;
+    if(!once) {
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: I think this condition should not be here anymore with the lastSrcId check above.

Comment (patrickelectric) [reply src/sensor/ping.cpp]: I did it, my I have done some ctrl+z and this got lost.

**Pull request merged**